tests/test_tokenization_common.py

```diff
@@ -1502,6 +1502,29 @@ def test_encode_plus_with_padding(self):
                     assert attention_mask + [0] * padding_size == right_padded_attention_mask
                     assert [0] * padding_size + attention_mask == left_padded_attention_mask
 
+    def test_right_and_left_truncation(self):
+        tokenizers = self.get_tokenizers(do_lower_case=False)
+        for tokenizer in tokenizers:
+            with self.subTest(f"{tokenizer.__class__.__name__}"):
+                sequence = "I wanted to go running"
+                max_length = 5
+
+                # LEFT TRUNCATION
+                tokenizer.truncation_side = "left"
+                truncated_sequence_left = tokenizer.encode(
+                    sequence,
+                    max_length=max_length,
+                    truncation=True,
+                )
+                truncated_sequence_left_length = len(truncated_sequence_left)
+                self.assertEqual(max_length, truncated_sequence_left_length)
+
+                # RIGHT TRUNCATION
+                tokenizer.truncation_side = "right"
+                truncated_sequence_right = tokenizer.encode(sequence, max_length=max_length, truncation=True)
+                truncated_sequence_right_length = len(truncated_sequence_right)
+                self.assertEqual(max_length, truncated_sequence_right_length)
```

Comment:
I propose in this test to also check that the encoding values correspond to the left or right values respectively. I propose here a solution without taking into account the special tokens, which seem to me more complicated to take into account automatically in the verification.

This test allows to observe that the new feature does not work correctly for the fast versions of tokenizers.

```suggestion
                sequence = "I wanted to go running, swimming and singing."
                max_length = 5

                tokenize_sequence = tokenizer(sequence, add_special_tokens=False)

                # LEFT TRUNCATION
                tokenizer.truncation_side = "left"
                truncated_sequence_left_without_special_tokens = tokenizer(
                    sequence, max_length=max_length, truncation=True, add_special_tokens=False
                )
                truncated_sequence_left_with_special_tokens = tokenizer(
                    sequence,
                    max_length=max_length,
                    truncation=True,
                )

                for key in tokenize_sequence.keys():
                    self.assertEqual(max_length, len(truncated_sequence_left_with_special_tokens[key]))
                    self.assertListEqual(
                        tokenize_sequence[key][-max_length:], truncated_sequence_left_without_special_tokens[key]
                    )

                # RIGHT TRUNCATION
                tokenizer.truncation_side = "right"
                truncated_sequence_right_without_special_tokens = tokenizer(
                    sequence, max_length=max_length, truncation=True, add_special_tokens=False
                )
                truncated_sequence_right_with_special_tokens = tokenizer(
                    sequence,
                    max_length=max_length,
                    truncation=True,
                )

                for key in tokenize_sequence.keys():
                    self.assertEqual(max_length, len(truncated_sequence_right_with_special_tokens[key]))
                    self.assertListEqual(
                        tokenize_sequence[key][:max_length], truncated_sequence_right_without_special_tokens[key]
                    )
```